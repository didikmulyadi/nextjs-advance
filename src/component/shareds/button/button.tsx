import './button.css'

import React from 'react'

/**
 * Button component props
 */
interface IButtonProps {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean
  /**
   * What background color to use
   */
  backgroundColor?: string
  /**
   * How large should the button be?
   */
  size?: 'small' | 'medium' | 'large'
  /**
   * Button contents
   */
  label: string
  /**
   * Optional click handler
   */
  onClick?: () => void
}

export const Button = (_props: IButtonProps) => {
  const {
    primary = false,
    size = 'medium',
    backgroundColor,
    label,
    ...props
  } = _props

  const mode = primary
    ? 'storybook-button--primary'
    : 'storybook-button--secondary'

  return (
    <button
      type="button"
      className={['storybook-button', `storybook-button--${size}`, mode].join(
        ' '
      )}
      {...props}
    >
      {label}
      <style jsx>{`
        button {
          background-color: ${backgroundColor};
        }
      `}</style>
    </button>
  )
}
