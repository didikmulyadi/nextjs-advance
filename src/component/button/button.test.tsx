import { render, screen } from '@testing-library/react'

import { Button } from './button'

describe('Button', () => {
  it('renders button unchanged', () => {
    const { container } = render(<Button label="save" />)
    expect(container).toMatchSnapshot()
  })

  it('renders a button', () => {
    render(<Button label="save" data-testid="button__save" />)

    const button = screen.getByTestId('button__save')

    expect(button).toBeInTheDocument()
    expect(
      button.classList.contains('storybook-button--secondary')
    ).toBeTruthy()
  })

  it('renders primary button correctly', () => {
    render(<Button label="save" data-testid="button__save" primary />)

    const button = screen.getByTestId('button__save')

    expect(button).toBeInTheDocument()
    expect(button.classList.contains('storybook-button--primary')).toBeTruthy()
  })
})
