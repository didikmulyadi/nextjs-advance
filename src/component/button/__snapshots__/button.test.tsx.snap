// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button unchanged 1`] = `
<div>
  <button
    class="jsx-781207911 storybook-button storybook-button--medium storybook-button--secondary"
    type="button"
  >
    save
  </button>
</div>
`;
